pkg/grapicmd/di/providers.go

```diff
@@ -19,57 +19,45 @@ import (
 )
 
 var (
-	ui   clui.UI
-	uiMu sync.Mutex
-
 	gexCfg   *gex.Config
 	gexCfgMu sync.Mutex
 
 	toolRepo   tool.Repository
 	toolRepoMu sync.Mutex
 )
 
-func ProvideUI(ctx *grapicmd.Ctx) clui.UI {
-	uiMu.Lock()
-	defer uiMu.Unlock()
-	if ui == nil {
-		ui = clui.New(ctx.OutWriter, ctx.InReader)
-	}
-	return ui
+func ProvideIO(ctx *grapicmd.Ctx) *cli.IO {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideIO should have comment or be unexported